README.md

```diff
@@ -351,11 +347,15 @@ which will perform a normal sysroot build, followed by a 'check' build of *your
   host's target triple with `rustc -vV`. On *nix, the following rune will extract the triple:
   `rustc -vV | egrep '^host: ' | sed 's/^host: //'`.
 
-- Remember that both `core` and `std` will get implicitly linked to your crate but *all the other
-  sysroot crates* will *not*. This means that if your Xargo.toml contains a crate like
-  `compiler_builtins` or `alloc` then you will have to add a `extern crate compiler_builtins` or
-  `extern crate alloc` *somewhere* in your dependency graph (either in your current crate or in some
-  of its dependencies).
+- Remember that `core`, `compiler_builtins` and `std` will get implicitly linked to your crate but
```

Comment:
Is `compiler_builtins` really implicitly linked (by the prelude, or so)? So, I can write `compiler_builtins::stuff` without having written `extern crate compiler_builtins`?

Reply:
You wrote

> Compiler builtins is automatically built and linked as part of libcore - it is generally unnecessary to specify it in the Xargo.toml.

And indeed compiler_builtins is added to the crate through some magic (I thought it was a dependency of "core" but it is not). However, the comment you added here is not about specifying it in `Xargo.toml`, it is about specifying it in the project that uses this sysroot. That doesn't entirely fit together, does it?

I thought this was about which things one can actually `use` in a crate that is built with this sysroot, but that is yet another question.

Reply:
Eeh, the way I understand it, "linking" is about binary-level linking. I've never heard of people talking about making crates available for `use` as "linking".

It's true, however, that for basically every other crate, there's no difference between the two. Compiler-builtins is the only weird crate being linked implicitly at the binary level, but not implicitly available for `use` at the rust source level. I don't think anyone ever does things like calling a `compiler_builtins` function manually, but still, in the strictest sense, it *may* be necessary to do `extern crate compiler_builtins` to pull it within scope... Thanks I hate it.

I guess I'll move compiler_builtins to its own section to make it clearer.

Reply:
In hindsight I think I was mostly confused by you talking about `compiler_builtins` appearing in `Xargo.toml` in your PR summary, which seems unrelated to this linking discussion (the discussion doesn't talk about `Xargo.toml` at all, after all).